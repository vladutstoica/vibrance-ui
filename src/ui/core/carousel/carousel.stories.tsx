import type { Meta, StoryObj } from "@storybook/react";

import React from "react";
import {
  Carousel,
  CarouselContent,
  CarouselItem,
  CarouselNext,
  CarouselPrevious,
} from "./carousel";

/**
 * A carousel with motion and swipe built using Embla.
 */
const meta: Meta<typeof Carousel> = {
  title: "Design System/Core/Carousel",
  component: Carousel,
  tags: ["autodocs"],
  argTypes: {},
  args: {
    className: "w-full max-w-xs",
  },
  render: (args) => (
    <Carousel {...args}>
      <CarouselContent>
        {Array.from({ length: 5 }).map((_, index) => (
          // biome-ignore lint/suspicious/noArrayIndexKey: <explanation>
          <CarouselItem key={index}>
            <div className="flex aspect-square items-center justify-center rounded border bg-card p-6">
              <span className="font-semibold text-4xl">{index + 1}</span>
            </div>
          </CarouselItem>
        ))}
      </CarouselContent>
      <CarouselPrevious />
      <CarouselNext />
    </Carousel>
  ),
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof Carousel>;

export default meta;

type Story = StoryObj<typeof meta>;

/**
 * The default form of the carousel.
 */
export const Default: Story = {};

/**
 * Use the `basis` utility class to change the size of the carousel.
 */
export const Size: Story = {
  render: (args) => (
    <Carousel {...args} className="mx-12 w-full max-w-xs">
      <CarouselContent>
        {Array.from({ length: 5 }).map((_, index) => (
          // biome-ignore lint/suspicious/noArrayIndexKey: <explanation>
          <CarouselItem key={index} className="basis-1/3">
            <div className="flex aspect-square items-center justify-center rounded border bg-card p-6">
              <span className="font-semibold text-4xl">{index + 1}</span>
            </div>
          </CarouselItem>
        ))}
      </CarouselContent>
      <CarouselPrevious />
      <CarouselNext />
    </Carousel>
  ),
  args: {
    className: "mx-12 w-full max-w-xs",
  },
};
