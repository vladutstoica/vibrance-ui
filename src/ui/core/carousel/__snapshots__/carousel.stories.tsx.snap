// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Core/Carousel Default smoke-test 1`] = `
<section class="relative w-full max-w-xs"
         aria-roledescription="carousel"
>
  <div class="overflow-hidden">
    <div class="flex -ml-4"
         style="transform: translate3d(0px, 0px, 0px);"
    >
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 basis-full pl-4"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            1
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 basis-full pl-4"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            2
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 basis-full pl-4"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            3
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 basis-full pl-4"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            4
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 basis-full pl-4"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            5
          </span>
        </div>
      </div>
    </div>
  </div>
  <button class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground absolute h-8 w-8 rounded-full -left-12 -translate-y-1/2 top-1/2"
          disabled
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="24"
         height="24"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="lucide lucide-arrow-left h-4 w-4"
    >
      <path d="m12 19-7-7 7-7">
      </path>
      <path d="M19 12H5">
      </path>
    </svg>
    <span class="sr-only">
      Previous slide
    </span>
  </button>
  <button class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground absolute h-8 w-8 rounded-full -right-12 -translate-y-1/2 top-1/2">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="24"
         height="24"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="lucide lucide-arrow-right h-4 w-4"
    >
      <path d="M5 12h14">
      </path>
      <path d="m12 5 7 7-7 7">
      </path>
    </svg>
    <span class="sr-only">
      Next slide
    </span>
  </button>
</section>
`;

exports[`Design System/Core/Carousel Size smoke-test 1`] = `
<section class="relative mx-12 w-full max-w-xs"
         aria-roledescription="carousel"
>
  <div class="overflow-hidden">
    <div class="flex -ml-4"
         style="transform: translate3d(0px, 0px, 0px);"
    >
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 pl-4 basis-1/3"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            1
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 pl-4 basis-1/3"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            2
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 pl-4 basis-1/3"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            3
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 pl-4 basis-1/3"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            4
          </span>
        </div>
      </div>
      <div role="group"
           aria-roledescription="slide"
           class="min-w-0 shrink-0 grow-0 pl-4 basis-1/3"
      >
        <div class="flex aspect-square items-center justify-center rounded border bg-card p-6">
          <span class="font-semibold text-4xl">
            5
          </span>
        </div>
      </div>
    </div>
  </div>
  <button class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground absolute h-8 w-8 rounded-full -left-12 -translate-y-1/2 top-1/2"
          disabled
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="24"
         height="24"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="lucide lucide-arrow-left h-4 w-4"
    >
      <path d="m12 19-7-7 7-7">
      </path>
      <path d="M19 12H5">
      </path>
    </svg>
    <span class="sr-only">
      Previous slide
    </span>
  </button>
  <button class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground absolute h-8 w-8 rounded-full -right-12 -translate-y-1/2 top-1/2">
    <svg xmlns="http://www.w3.org/2000/svg"
         width="24"
         height="24"
         viewbox="0 0 24 24"
         fill="none"
         stroke="currentColor"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="lucide lucide-arrow-right h-4 w-4"
    >
      <path d="M5 12h14">
      </path>
      <path d="m12 5 7 7-7 7">
      </path>
    </svg>
    <span class="sr-only">
      Next slide
    </span>
  </button>
</section>
`;
